proto/connectrpc/conformance/v1alpha1/service.proto

```diff
@@ -60,66 +73,116 @@ service ConformanceService {
   // If the input stream is empty, the server should send a single response
   // message that includes no data and only the request properties (headers,
   // timeout).
-  rpc BidiStream(stream BidiStreamRequest) returns (stream ConformancePayload);
+  //
+  // Note that clients could specify a list of desired responses that is less than
+  // or greater than the amount of requests they send on the stream. If this is the case
+  // and full duplex is specified then servers should behave as follows:
+  //
+  // If the total desired responses is less than the requests, the server should
+  // simply stop sending responses once the total desired is sent and continue to read
+  // messages until the client is finished sending.
+  //
+  // If the total desired responses is greater than the requests, the server should
+  // flush the remaining responses once the client is finished sending.
```

Comment:
This is up for debate probably. Should we allow a number of desired responses that is unequal to the requests sent by the client for full-duplex calls? It seems like we probably should, but I'm open to suggestions otherwise.

Reply:
> // If the total desired responses is less than the requests, the server should
>  // simply stop sending responses once the total desired is sent and continue to read
>  // messages until the client is finished sending.

Might this cause the test to deadlock? If the client-under-test did the naive thing here and sends one request, waits for one response, etc, then after the last request, it will hang waiting on a response. I think this is a case where it's probably better to abort with an error if we get a message and have no corresponding response to send instead of risking a deadlock.

As far as what to do if number of responses > number of requests, looks good. In practice, the test cases will not have these kinds of anomalous configuration.

Reply:
Fair point. I just removed all these instructions from the comments since the test cases won't be constructed this way.